File: pdu/python/builtin_interfaces/pdu_cdr_conv_Time.py
from .pdu_pytype_Time import Time
from ..pdu_cdr_runtime import CdrReader, CdrWriter




def py_to_cdr_body_Time(writer: CdrWriter, src: Time):
    writer.write_int32(src.sec)
    writer.write_uint32(src.nanosec)


def cdr_body_to_py_Time(reader: CdrReader, dst: Time):
    dst.sec = reader.read_int32()
    dst.nanosec = reader.read_uint32()
    return dst


def py_to_cdr_Time(src: Time) -> bytes:
    writer = CdrWriter()
    writer.write_encapsulation()
    py_to_cdr_body_Time(writer, src)
    return writer.bytes()


def cdr_to_py_Time(cdr_payload) -> Time:
    reader = CdrReader(cdr_payload)
    reader.read_encapsulation()
    dst = Time()
    return cdr_body_to_py_Time(reader, dst)

File: pdu/javascript/builtin_interfaces/pdu_cdr_conv_Time.js
import { Time } from './pdu_jstype_Time.js';
import { PduCdrWriter, PduCdrReader } from '../pdu_cdr_runtime.js';


export class PduTimeConverter {
    /**
     * @param {PduCdrWriter} writer
     * @param { Time } src
     */
    static to_cdr_body(writer, src) {
        writer.write_int32(src.sec);
        writer.write_uint32(src.nanosec);
    }

    /**
     * @param {PduCdrReader} reader
     * @param { Time } dst
     * @returns { Time }
     */
    static cdr_body_to_js(reader, dst) {
        dst.sec = reader.read_int32();
        dst.nanosec = reader.read_uint32();
        return dst;
    }

    /**
     * @param { Time } src
     * @returns {ArrayBuffer}
     */
    static to_cdr(src) {
        const writer = new PduCdrWriter();
        writer.write_encapsulation();
        this.to_cdr_body(writer, src);
        return writer.get_buf();
    }

    /**
     * @param {ArrayBuffer|ArrayBufferView} cdrPayload
     * @returns { Time }
     */
    static from_cdr(cdrPayload) {
        const reader = new PduCdrReader(cdrPayload);
        reader.read_encapsulation();
        return this.cdr_body_to_js(reader, new Time());
    }
}
